qiskit_dynamics/models/lindblad_model.py

```diff
@@ -0,0 +1,616 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Lindblad models module.
+"""
+
+from typing import Tuple, Union, List, Optional
+from scipy.sparse.csr import csr_matrix
+
+from qiskit import QiskitError
+from qiskit.quantum_info.operators import Operator
+from qiskit_dynamics.dispatch import Array
+from qiskit_dynamics.type_utils import to_numeric_matrix_type
+from qiskit_dynamics.signals import Signal, SignalList
+from .generator_model import BaseGeneratorModel, GeneratorModel
+from .hamiltonian_model import HamiltonianModel, is_hermitian
+from .operator_collections import (
+    BaseLindbladOperatorCollection,
+    DenseLindbladCollection,
+    DenseVectorizedLindbladCollection,
+    SparseLindbladCollection,
+    SparseVectorizedLindbladCollection,
+)
+from .rotating_frame import RotatingFrame
+
+
+class LindbladModel(BaseGeneratorModel):
+    r"""A model of a quantum system in terms of the Lindblad master equation.
+
+    The Lindblad master equation models the evolution of a density matrix according to:
+
+    .. math::
+        \dot{\rho}(t) = -i[H(t), \rho(t)] + \mathcal{D}(t)(\rho(t)),
+
+    where :math:`\mathcal{D}(t)` is the dissipator portion of the equation,
+    given by
+
+    .. math::
+        \mathcal{D}(t)(\rho(t)) = \sum_j \gamma_j(t) L_j \rho L_j^\dagger
+                                  - \frac{1}{2} \{L_j^\dagger L_j, \rho\},
+
+    with :math:`[\cdot, \cdot]` and :math:`\{\cdot, \cdot\}` the
+    matrix commutator and anti-commutator, respectively. In the above:
+
+        - :math:`H(t)` denotes the Hamiltonian,
+        - :math:`L_j` denotes the :math:`j^{th}` dissipator, or Lindblad,
+          operator, and
+        - :math:`\gamma_j(t)` denotes the signal corresponding to the
+          :math:`j^{th}` Lindblad operator.
+
+    Instantiating an instance of :class:`~qiskit_dynamics.models.LindbladModel`
+    requires specifying the above decomposition:
+
+    .. code-block:: python
+
+        lindblad_model = LindbladModel(static_hamiltonian,
```

Comment:
Might be good to use kwargs instead of args in docs here too

Reply:
Agreed, changed.